File: JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands2/lobby/career/CreateUserResponse.java
package bluffinmuffin.protocol.commands2.lobby.career;

import java.util.StringTokenizer;

import bluffinmuffin.protocol.commands2.lobby.AbstractLobbyResponse;

public class CreateUserResponse extends AbstractLobbyResponse<CreateUserCommand>
{
    @Override
    protected String getCommandName()
    {
        return CreateUserResponse.COMMAND_NAME;
    }
    
    public static String COMMAND_NAME = "lobbyCAREER_CREATE_USER_RESPONSE";
    private final boolean m_isOK;
    
    public CreateUserResponse(StringTokenizer argsToken)
    {
        super(new CreateUserCommand(argsToken));
        m_isOK = Boolean.parseBoolean(argsToken.nextToken());
    }
    
    public CreateUserResponse(CreateUserCommand command, boolean ok)
    {
        super(command);
        m_isOK = ok;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        super.encode(sb);
        append(sb, m_isOK);
    }
    
    public boolean isOK()
    {
        return m_isOK;
    }
}

File: trunk/C#/Com.Ericmas001.Game.Poker.Protocol.Commands/Lobby/Career/CreateUserResponse.cs
namespace Com.Ericmas001.Game.Poker.Protocol.Commands.Lobby.Career
{
    public class CreateUserResponse : AbstractLobbyResponse<CreateUserCommand>
    {
        public bool Success { get; set; }

        public CreateUserResponse()
        {
        }

        public CreateUserResponse(CreateUserCommand command)
            : base(command)
        {
        }
    }
}
